.github/workflows/tests.yaml

```diff
@@ -48,15 +48,74 @@ jobs:
           name: vault-csi-provider-image
           path: ${{ env.TARBALL_FILE }}
 
+  vault-versions:
+    runs-on: ubuntu-latest
+    steps:
+      - run: echo "setting vault versions"
+    outputs:
```

Comment:
I don't mind this way, but the top-level `env` would probably be a better fit for setting these workflow constants instead of running a dedicated step to get outputs, ref: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#env. You would then refer to the values similarly below as `${{ env.VAULT_N }}` if you need to use GitHub workflows interpolation to make it part of the step definition, or obviously you can also use the normal shell env syntax within shell scripts.

The main benefit you'd get is a bit of simplification because you no longer need to manage `needs` entries for these constants.

Reply:
IIRC we ran into trouble referring to top-level `env`'s in the matrix definition while setting this up for the operator. Maybe something's changed since then?

Reply:
Oops sorry you're right, it seems env values can only be interpolated in steps, not jobs. The docs aren't terribly explicit about the jobs side of that, but here's the reference: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#env